### PR TITLE
Add examples for `doctest` to `optuna/prunes/*.py`

### PR DESCRIPTION
**Pull request by nuka137**
This PR add doctest example to below files.

* optuna/pruners/successive_halving
* optuna/pruners/nop
* optuna/pruners/median
* optuna/pruners/percentile

Related Issue: #734

### CODE REVIEW

**Comment by hvy** [comment]
Thanks @nuka137 for your constant contributions. We'll review it as soon as possible.

File: optuna/pruners/nop.py [hunk 1]
```diff
@@ -11,16 +11,42 @@ class NopPruner(BasePruner):
...
+                n_train_iter = 100
+
+                for step in range(n_train_iter):
+                    clf.partial_fit(X_train, y_train, classes=classes)
+
+                    intermediate_value = clf.score(X_test, y_test)
+                    trial.report(intermediate_value, step)
+
+                    if trial.should_prune():
+                        print("Never reach here")
```
Comment (hvy) [inline optuna/pruners/nop.py]: How about a slightly stricter doctest?
```suggestion
                    assert False, "should_prune() should always return False with this pruner."
```

**Comment by nuka137** [comment]
@hvy 

Thanks for your review!
It's a good idea to apply strict doctest for `NopPruner`.
I fixed it.

**Comment by hvy** [comment]
If duplication is a concern, we could also not have any example code at all. Instead, we could in the documentation for each pruner just refer to the example in the `TrialPruned` exception with a `.. seealso::`. In addition, we could improve the `TrialPruned` exception example to specify a particular pruner instead of relying on the default (https://github.com/nuka137/optuna/blob/master/optuna/exceptions.py#L48) to make it clearer.

**Comment by toshihikoyanase** [comment]
@hvy Thank you for your suggestion. It sounds good to me, but it still has some cons (e.g., users need one more click to see the example).

So, how about accepting this PR for now and create a new PR if we work on it?

**Pull request merged**